src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -202,13 +207,19 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     maxBandwidthKbps: number
   ): void {
     let dimension = this.browserBehavior.requiresResolutionAlignment(width, height);
-    this.videoWidth = dimension[0];
-    this.videoHeight = dimension[1];
-    this.videoFrameRate = frameRate;
-    this.videoMaxBandwidthKbps = maxBandwidthKbps;
+    this.videoInputQualitySettings = new VideoQualitySettings(
+      dimension[0],
+      dimension[1],
+      frameRate,
+      maxBandwidthKbps
+    );
     this.updateMaxBandwidthKbps();
   }
 
+  getVideoInputQualitySettings(): VideoQualitySettings {
```

Comment:
How about removing `Settings` to match the setter `chooseVideoInputQuality`? We don't have other model classes ending with `Settings`. But I don't have a strong preference.

Reply:
NVM. I think `Settings` is okay. I found that `MediaTrackSettings` includes the same properties in Web API.